api/service/synchronize/stagedstreamsync/stage_bodies.go

```diff
@@ -469,14 +469,27 @@ func (b *StageBodies) verifyBlockAndExtractReceiptsData(batchBlockBytes [][]byte
 		if blockBytes == nil {
 			continue
 		}
-		if err := rlp.DecodeBytes(blockBytes, &block); err != nil {
-			b.configs.logger.Error().
-				Uint64("block number", i).
-				Msg("block size invalid")
-			return ErrInvalidBlockBytes
-		}
-		if sigBytes != nil {
-			block.SetCurrentCommitSig(sigBytes)
+		var err error
```

Comment:
we already have function for this `RlpDecodeBlockOrBlockWithSig`

Reply:
good suggestion, have updated it